### ISSUE TITLE
Unstable carbon instance

### ISSUE THREAD

**Issue by kira510** [graphite-project/carbon#652]
Hello,
I have 3 graphite servers which were working fine but recently we have had stability issues with one of the nodes.

I see this error in the console.log which occurs repeatedly,

```
04/05/2017 06:10:35 :: Starting factory CarbonClientFactory(127.0.0.1:2024:2)
04/05/2017 06:10:35 :: <twisted.internet.tcp.Connector instance at 0x7ff93269fea8> will retry in 4 seconds
04/05/2017 06:10:35 :: Stopping factory CarbonClientFactory(127.0.0.1:2024:2)
04/05/2017 06:10:40 :: Starting factory CarbonClientFactory(127.0.0.1:2024:2)
04/05/2017 06:18:58 :: <twisted.internet.tcp.Connector instance at 0x7ff9326af290> will retry in 3 seconds
04/05/2017 06:18:58 :: Stopping factory CarbonClientFactory(127.0.0.1:2014:1)
04/05/2017 06:19:01 :: Starting factory CarbonClientFactory(127.0.0.1:2014:1)
04/05/2017 06:19:01 :: <twisted.internet.tcp.Connector instance at 0x7ff9326af290> will retry in 5 seconds
04/05/2017 06:19:01 :: Stopping factory CarbonClientFactory(127.0.0.1:2014:1)
04/05/2017 06:19:03 :: Exception in cacheFull event handler: args=() kwargs={}
Traceback (most recent call last):
  File "/opt/owl/graphite/lib/twisted/internet/defer.py", line 393, in callback
    self._startRunCallbacks(result)
  File "/opt/owl/graphite/lib/twisted/internet/defer.py", line 501, in _startRunCallbacks
    self._runCallbacks()
  File "/opt/owl/graphite/lib/twisted/internet/defer.py", line 588, in _runCallbacks
    current.result = callback(current.result, *args, **kw)
  File "/opt/owl/graphite/lib/carbon/client.py", line 114, in queueFullCallback
    events.cacheFull()
--- <exception caught here> ---
  File "/opt/owl/graphite/lib/carbon/events.py", line 20, in __call__
    handler(*args, **kwargs)
  File "/opt/owl/graphite/lib/carbon/events.py", line 32, in <lambda>
    cacheFull.addHandler(lambda: carbon.instrumentation.increment('cache.overflow'))
exceptions.NameError: global name 'carbon' is not defined

04/05/2017 06:19:07 :: Starting factory CarbonClientFactory(127.0.0.1:2014:1)
04/05/2017 06:19:07 :: <twisted.internet.tcp.Connector instance at 0x7ff9326af290> will retry in 5 seconds
04/05/2017 06:19:07 :: Stopping factory CarbonClientFactory(127.0.0.1:2014:1)
04/05/2017 06:19:13 :: Starting factory CarbonClientFactory(127.0.0.1:2014:1)
04/05/2017 06:19:13 :: <twisted.internet.tcp.Connector instance at 0x7ff9326af290> will retry in 6 seconds
04/05/2017 06:19:13 :: Stopping factory CarbonClientFactory(127.0.0.1:2014:1)
04/05/2017 06:19:19 :: Starting factory CarbonClientFactory(127.0.0.1:2014:1)
04/05/2017 06:19:19 :: <twisted.internet.tcp.Connector instance at 0x7ff9326af290> will retry in 4 seconds
```
Also i see these error in the clients.log,
````
04/05/2017 06:35:58 :: CarbonClientFactory(127.0.0.1:2014:1)::clientConnectionFailed (127.0.0.1:2014) Connection was refused by other side: 111: Connection refused.
04/05/2017 06:36:02 :: CarbonClientFactory(127.0.0.1:2014:1)::startedConnecting (127.0.0.1:2014)
04/05/2017 06:36:02 :: CarbonClientFactory(127.0.0.1:2014:1)::clientConnectionFailed (127.0.0.1:2014) Connection was refused by other side: 111: Connection refused.
04/05/2017 06:36:06 :: CarbonClientFactory(127.0.0.1:2014:1)::startedConnecting (127.0.0.1:2014)
````
Please help, and please explain clearly the issue.

**Comment by deniszh** [comment]
@kira510 - that's a bug, it was fixed in 0.9.16
In your case it's just a symptom though, your cache is overflowing, and throwing this error trying to instrument that condition.

**Comment by kira510** [comment]
Alright, thanks a lot. Upgrading over 0.9.13 might not fix the issue, i will perform a fresh deployment of 0.9.16
